PWGHF/TableProducer/candidateCreatorSigmac0plusplusCascade.cxx

```diff
@@ -0,0 +1,324 @@
+  Configurable<double> cfgtrkMaxEta{"cfgtrkMaxEta", 0.8, "track max Eta"};
+  Configurable<double> cfgMaxDCAxyToPVcut{"cfgMaxDCAxyToPVcut", 2.0, "Track DCAxy cut to PV Maximum"};
+  Configurable<double> cfgMaxDCAzToPVcut{"cfgMaxDCAzToPVcut", 2.0, "Track DCAz cut to PV Maximum"};
+  Configurable<double> cfgnFindableTPCClusters{"cfgnFindableTPCClusters", 120, "nFindable TPC Clusters"};
+  Configurable<double> cfgnTPCCrossedRows{"cfgnTPCCrossedRows", 70, "nCrossed TPC Rows"};
+  Configurable<double> cfgnTPCChi2{"cfgnTPChi2", 4.0, "nTPC Chi2 per Cluster"};
+  Configurable<double> cfgnITSChi2{"cfgnITShi2", 36.0, "nITS Chi2 per Cluster"};
+  Configurable<double> cfgTPCnSigmaPi{"cfgTPCnSigmaPi", 3.0, "TPC nSigma selection"};
+
+  /// Table with Σc0,++ info
+  Produces<aod::HfCandScCasBase> rowScCandidates;
+  /// Selection of candidates Λc+
+  Configurable<int> selectionFlagLc{"selectionFlagLc", 1, "Selection Flag for Lc"};
+  Configurable<double> yCandLcMax{"yCandLcMax", -1., "max. candLc. Lc rapidity"};
+  Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{hf_cuts_sigmac_to_p_k_pi::vecBinsPt}, "pT bin limits"};
+  Configurable<LabeledArray<double>> cutsMassLcMax{"cutsMassLcMax", {hf_cuts_sigmac_to_p_k_pi::cuts[0], hf_cuts_sigmac_to_p_k_pi::nBinsPt, hf_cuts_sigmac_to_p_k_pi::nCutVars, hf_cuts_sigmac_to_p_k_pi::labelsPt, hf_cuts_sigmac_to_p_k_pi::labelsCutVar}, "Lc candidate selection per pT bin"};
+  Configurable<int> selectionFlagLcToK0sP{"selectionFlagLcToK0sP", 1, "Selection Flag for Lc"};
+  Configurable<int> selectionFlagLcbarToK0sP{"selectionFlagLcbarToK0sP", 1, "Selection Flag for Lcbar"};
+  /// Selections on candidate soft π-,+
+  Configurable<bool> applyGlobalTrkWoDcaCutsSoftPi{"applyGlobalTrkWoDcaCutsSoftPi", false, "Switch on the application of the global-track w/o dca cuts for soft pion BEFORE ALL OTHER CUSTOM CUTS"};
+  Configurable<float> softPiEtaMax{"softPiEtaMax", 0.9f, "Soft pion max value for pseudorapidity (abs vale)"};
+  Configurable<float> softPiChi2Max{"softPiChi2Max", 36.f, "Soft pion max value for chi2 ITS"};
+  Configurable<int> softPiItsHitMap{"softPiItsHitMap", 127, "Soft pion ITS hitmap"};
+  Configurable<int> softPiItsHitsMin{"softPiItsHitsMin", 1, "Minimum number of ITS layers crossed by the soft pion among those in \"softPiItsHitMap\""};
+  Configurable<float> softPiDcaXYMax{"softPiDcaXYMax", 0.065, "Soft pion max dcaXY (cm)"};
+  Configurable<float> softPiDcaZMax{"softPiDcaZMax", 0.065, "Soft pion max dcaZ (cm)"};
+  Configurable<bool> addQA{"addQA", true, "Switch for the qa PLOTS"};
+
+  template <typename TrackType>
+  bool isTrackSelected(const TrackType& track)
+  {
+    if (track.pt() < cfgtrkMinPt)
+      return false;
+    if (std::abs(track.eta()) > cfgtrkMaxEta)
+      return false;
+    if (std::abs(track.dcaXY()) > cfgMaxDCAxyToPVcut)
+      return false;
+    if (std::abs(track.dcaZ()) > cfgMaxDCAzToPVcut)
+      return false;
```

Comment:
A warning for you: this works if and only if you do not use the time-to-collision reassociation of soft pion tracks. In case you want to do it, instead, you should repropagate the soft pion track to the new PV the track is re-associated to, see here as an example https://github.com/AliceO2Group/O2Physics/blob/master/PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx#L274-L294
It depends what you want to do. I did some tests in the past and using or not this reassociation does not seem to change significantly the things, so probably it's not a priority at this stage.

Reply:
Thank you very much for the relevant information. Currently, I am not using the time-to-collision-association. If I will use it in future, I will consider this.